fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluids/v1/container/SimpleFluidContainer.java

```diff
@@ -0,0 +1,113 @@
+package net.fabricmc.fabric.api.fluids.v1.container;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import com.google.common.collect.Iterables;
+
+import net.minecraft.fluid.Fluid;
+
+import net.fabricmc.fabric.Action;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.FluidVolume;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.ImmutableFluidVolume;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.SimpleFluidVolume;
+
+public class SimpleFluidContainer implements FluidContainer {
+	private final Iterable<FluidContainer> containers;
+
+	public SimpleFluidContainer(FluidContainer... containers) {
+		this(Arrays.asList(containers));
+	}
+
+	public SimpleFluidContainer(Iterable<FluidContainer> containers) {
+		this.containers = containers;
+	}
+
+	@Override
+	public FluidVolume drain(Fluid fluid, long amount, Action simulate) {
+		FluidVolume current = new SimpleFluidVolume();
```

Comment:
Hmm, I want someone else to look into this, since you never receive your container back (contractually immutable)

Reply:
u receive your own container back if it's not modified or if it's not a simulation